equinix/resource_metal_device_acc_test.go

```diff
@@ -625,6 +706,41 @@ resource "equinix_metal_device" "test" {
 `, confAccMetalDevice_base(preferable_plans, preferable_metros, preferable_os), projSuffix, rInt, rInt, testDeviceTerminationTime())
 }
 
+func testAccMetalDeviceConfig_allowAttributeChanges(rInt int, projSuffix string, userdata string, customdata string, attributeName string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "equinix_metal_project" "test" {
+    name = "tfacc-device-%s"
+}
+
+resource "equinix_metal_device" "test" {
+  hostname         = "tfacc-test-device-%d"
+  plan             = local.plan
+  metro            = local.metro
+  operating_system = local.os
+  billing_cycle    = "hourly"
+  project_id       = "${equinix_metal_project.test.id}"
+  tags             = ["%d"]
+  user_data        = %q
+  custom_data      = %q
+
```

Comment:
most (all?) of the metal_device tests include `termination_time` as a stop-gap to flush out test instances in the event that E2E sweeping doesn't pick the devices up.